### PR TITLE
Feature 137 sig in with existing user

### PR DESCRIPTION
**Pull request by JustTheDude001**
Changes With Main Merged into feature #137.
 .env                                                                                                                                 |    11 +
 .env.docker                                                                                                                          |     4 +
 .github/CODEOWNERS                                                                                                                   |     3 +
 .github/workflows/{pull_request_phpunit_tests.yml => backend_check.yml}                                                              |    10 +-
 .github/workflows/deploy.yml                                                                                                         |     2 +-
 .github/workflows/frontend_check.yml                                                                                                 |    58 +
 .gitignore                                                                                                                           |     1 -
 Makefile                                                                                                                             |     2 +
 README.md                                                                                                                            |   135 +-
 .../OpenApi/controllersAnnotations/{studentResumeAnnotations => additionalTrainingAnnotations}/AnnotationsAdditionalTraining.php     |    22 +-
 app/Annotations/OpenApi/controllersAnnotations/adminAnnotations/AnnotationsAdmin.php                                                 |   285 ++
 app/Annotations/OpenApi/controllersAnnotations/authAnnotations/AnnotationsAuth.php                                                   |   159 ++
 app/Annotations/OpenApi/controllersAnnotations/{tagAnnotations => developmentAnnotations}/AnnotationsDevelopmentList.php             |     8 +-
 app/Annotations/OpenApi/controllersAnnotations/{studentResumeAnnotations => modalityAnnotations}/AnnotationsModality.php             |    13 +-
 app/Annotations/OpenApi/controllersAnnotations/recruiterAnnotations/AnnotationsRecruiter.php                                         |   257 ++
 .../OpenApi/controllersAnnotations/{studentResumeAnnotations => specializationListAnnotations}/AnnotationsSpecializationList.php     |     4 +-
 app/Annotations/OpenApi/controllersAnnotations/studentAnnotations/AnnotationsStudent.php                                             |   295 ++
 app/Annotations/OpenApi/controllersAnnotations/{studentResumeAnnotations => studentAnnotations}/AnnotationsStudentDetail.php         |    15 +-
 app/Annotations/OpenApi/controllersAnnotations/{studentResumeAnnotations => studentAnnotations}/AnnotationsStudentList.php           |    14 +-
 .../controllersAnnotations/{studentResumeAnnotations => studentBootcampDetailAnnotations}/AnnotationsStudentBootcampDetail.php       |    14 +-
 .../OpenApi/controllersAnnotations/{studentResumeAnnotations => studentCollaborationAnnotations}/AnnotationsStudentCollaboration.php |    13 +-
 .../controllersAnnotations/{studentResumeAnnotations => studentLanguagesDetailAnnotations}/AnnotationsStudentLanguagesDetail.php     |    12 +-
 .../controllersAnnotations/{studentResumeAnnotations => studentProjectsDetailAnnotations}/AnnotationsStudentProjectsDetail.php       |    16 +-
 app/Annotations/OpenApi/controllersAnnotations/tagAnnotations/AnnotationsTag.php                                                     |    54 +-
 app/Annotations/OpenApi/modelsAnnotations/studentAnnotations/AnnotationsStudent.php                                                  |     3 +-
 app/DTO/UserDTO.php                                                                                                                  |    15 +
 app/Http/Controllers/api/AuthController.php                                                                                          |    61 +
 app/Http/Requests/SigninRequest.php                                                                                                  |    38 +
 app/Service/User/UserService.php                                                                                                     |   123 +
 composer.json                                                                                                                        |     1 +
 composer.lock                                                                                                                        |   655 +++--
 config/app.php                                                                                                                       |     1 +
 config/database.php                                                                                                                  |    22 +
 database/seeders/DatabaseSeeder.php                                                                                                  |     2 +
 database/seeders/SigninTestSeeder.php                                                                                                |    40 +
 db.json                                                                                                                              |    13 +
 docker-compose.yml                                                                                                                   |    15 +-
 docker/.gitignore                                                                                                                    |     1 +
 docker/redis/.conf/redis.conf                                                                                                        |  1320 +++++++++
 index.html                                                                                                                           |    13 +
 package-lock.json                                                                                                                    | 10273 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-----
 package.json                                                                                                                         |    72 +-
 postcss.config.js                                                                                                                    |     6 +
 public/vite.svg                                                                                                                      |     1 +
 routes/api/v1.php                                                                                                                    |   123 +-
 sonar-project.properties                                                                                                             |     9 +
 src/App.tsx                                                                                                                          |    11 +
 src/__tests__/components/Button.test.tsx                                                                                             |    19 +
 src/__tests__/components/MenuNavbar.test.tsx                                                                                         |    27 +
 src/__tests__/components/StudentDetailsContent.test.tsx                                                                              |    33 +
 src/__tests__/components/studentDetailCards/LanguageCard.test.tsx                                                                    |    68 +
 src/__tests__/components/studentFilters/StudentFiltersContent.test.tsx                                                               |    36 +
 src/__tests__/components/students/StudentCard.test.tsx                                                                               |    38 +
 src/__tests__/context/StudentIdContext.test.tsx                                                                                      |    30 +
 src/__tests__/setup.ts                                                                                                               |    17 +
 src/__tests__/test-utils.tsx                                                                                                         |    22 +
 src/api/FetchStudentsList.ts                                                                                                         |    26 +
 src/assets/img/arrow-right-projects.png                                                                                              |   Bin 0 -> 491 bytes
 src/assets/img/folder.png                                                                                                            |   Bin 0 -> 3458 bytes
 src/assets/img/index.ts                                                                                                              |     6 +
 src/assets/img/mail-gray.png                                                                                                         |   Bin 0 -> 884 bytes
 src/assets/img/medal-dynamic-color.png                                                                                               |   Bin 0 -> 5047 bytes
 src/assets/img/stud_1.png                                                                                                            |   Bin 0 -> 10933 bytes
 src/assets/img/stud_2.png                                                                                                            |   Bin 0 -> 10550 bytes
 src/assets/img/stud_3.png                                                                                                            |   Bin 0 -> 10466 bytes
 src/assets/img/target.png                                                                                                            |   Bin 0 -> 4168 bytes
 src/assets/svg/arrow-down.svg                                                                                                        |     3 +
 src/assets/svg/arrow-left.svg                                                                                                        |     3 +
 src/assets/svg/arrow-right.svg                                                                                                       |     3 +
 src/assets/svg/bookmark.svg                                                                                                          |     3 +
 src/assets/svg/burgerMenu.svg                                                                                                        |     3 +
 src/assets/svg/close.svg                                                                                                             |     4 +
 src/assets/svg/email.svg                                                                                                             |     3 +
 src/assets/svg/github.svg                                                                                                            |     3 +
 src/assets/svg/index.ts                                                                                                              |    31 +
 src/assets/svg/itAcademyLogo.svg                                                                                                     |     9 +
 src/assets/svg/linkedin.svg                                                                                                          |     9 +
 src/assets/svg/lock-dynamic-color.svg                                                                                                |    23 +
 src/assets/svg/pencil.svg                                                                                                            |     3 +
 src/assets/svg/remoto.svg                                                                                                            |     3 +
 src/assets/svg/three-dots.svg                                                                                                        |     3 +
 src/assets/svg/userIcon.svg                                                                                                          |     4 +
 src/components/CheckTranslation.tsx                                                                                                  |    38 +
 src/components/SwitchTheme.tsx                                                                                                       |    38 +
 src/components/atoms/Button.tsx                                                                                                      |    31 +
 src/components/landing/Landing.tsx                                                                                                   |    27 +
 src/components/login_&_register/LoginButton.tsx                                                                                      |    15 +
 src/components/login_&_register/LoginPopup.tsx                                                                                       |    81 +
 src/components/login_&_register/RegisterPopup.tsx                                                                                    |   186 ++
 src/components/login_&_register/RestrictedAccessPopup.tsx                                                                            |   121 +
 src/components/menuNavbar/MenuNavbar.tsx                                                                                             |    46 +
 src/components/studentDetail/StudentDetailsContent.tsx                                                                               |    54 +
 src/components/studentDetail/StudentDetailsLayout.tsx                                                                                |    44 +
 src/components/studentDetailCards/bootcampSection/BootcampCard.tsx                                                                   |    20 +
 src/components/studentDetailCards/collaborationSection/ChallengesCard.tsx                                                            |    16 +
 src/components/studentDetailCards/collaborationSection/CollaborationCard.tsx                                                         |    14 +
 src/components/studentDetailCards/collaborationSection/ResourcesCard.tsx                                                             |    16 +
 src/components/studentDetailCards/languagesSection/LanguagesCard.tsx                                                                 |    50 +
 src/components/studentDetailCards/modalitySection/modalityCard.tsx                                                                   |    12 +
 src/components/studentDetailCards/otherEducationSection/OtherEducationCard.tsx                                                       |    42 +
 src/components/studentDetailCards/projectsSection/ProjectsCard.tsx                                                                   |   105 +
 src/components/studentDetailCards/studentDataSection/StudentDataCard.tsx                                                             |   130 +
 src/components/studentFilters/StudentFiltersContent.tsx                                                                              |   104 +
 src/components/studentFilters/StudentFiltersLayout.tsx                                                                               |     9 +
 src/components/studentFilters/StudentFiltersModal.tsx                                                                                |    29 +
 src/components/students/StudentCard.tsx                                                                                              |    67 +
 src/components/students/StudentsLayout.tsx                                                                                           |    28 +
 src/components/students/StudentsList.tsx                                                                                             |    47 +
 src/components/userNavBar/UserNavbar.tsx                                                                                             |   119 +
 src/context/SmallScreenContext.tsx                                                                                                   |    22 +
 src/context/StudentFiltersContext.tsx                                                                                                |    33 +
 src/context/StudentIdContext.tsx                                                                                                     |    41 +
 src/context/ThemeContext.tsx                                                                                                         |    12 +
 src/context/ThemeProvider.tsx                                                                                                        |    21 +
 src/data/profiles.ts                                                                                                                 |    77 +
 src/hooks/ReduxHooks.ts                                                                                                              |    22 +
 src/interfaces/interfaces.ts                                                                                                         |    39 +
 src/lib/const/endpoints.ts                                                                                                           |     7 +
 src/lib/data/supportedLngs.ts                                                                                                        |    20 +
 app/Http/Controllers/api/LoginController.php => src/lib/utils/.gitkeep                                                               |     0
 src/locales/i18n.ts                                                                                                                  |    32 +
 src/locales/languages/ca/ca.json                                                                                                     |     6 +
 src/locales/languages/en/en.json                                                                                                     |     6 +
 src/locales/languages/es/es.json                                                                                                     |     6 +
 src/main.tsx                                                                                                                         |    23 +
 src/pages/Home.tsx                                                                                                                   |     5 +
 src/schemes/schemas.ts                                                                                                               |    34 +
 src/store/reducers/getUserDetail/apiGetUserDetail.ts                                                                                 |    24 +
 src/store/store.ts                                                                                                                   |    13 +
 src/styles/App.css                                                                                                                   |    23 +
 src/styles/index.css                                                                                                                 |    10 +
 src/styles/normalize.css                                                                                                             |   352 +++
 src/vite-env.d.ts                                                                                                                    |     1 +
 storage/api-docs/api-docs.json                                                                                                       |  1421 ----------
 tailwind.config.js                                                                                                                   |    47 +
 tests/Feature/AuthControllerTest.php                                                                                                 |   108 +
 tests/Feature/SigninMockeryTest.php                                                                                                  |   230 ++
 tests/Feature/SigninTest.php                                                                                                         |   118 +
 tests/Feature/Student/AdditionalTrainingListControllerTest.php                                                                       |     6 +-
 tests/Feature/Student/CollaborationControllerTest.php                                                                                |     6 +-
 tests/Feature/Student/ModalityTest.php                                                                                               |     2 +-
 tests/Feature/Student/StudentBootcampDetailControllerTest.php                                                                        |     6 +-
 tests/Feature/Student/StudentDetailControllerTest.php                                                                                |     4 +-
 tests/Feature/Student/StudentLanguagesDetailControllerTest.php                                                                       |     8 +-
 tests/Feature/Student/StudentListTest.php                                                                                            |    10 +-
 tests/Feature/Student/StudentProjectsDetailControllerTest.php                                                                        |     8 +-
 tests/Feature/Student/StudentsListTest.php                                                                                           |    30 +
 tests/Feature/TagControllerTest.php                                                                                                  |    70 +
 tests/Unit/Service/UserServiceTest.php                                                                                               |   291 ++
 tsconfig.json                                                                                                                        |    23 +
 tsconfig.node.json                                                                                                                   |     9 +
 vite.config.ts                                                                                                                       |    20 +
 152 files changed, 16792 insertions(+), 2600 deletions(-)

In order to do not have the main merged checkout before this commit:
mmit 672490ab2d8f974bd8e10fcb34b164736e7cf9ca
Author: DudiDude <acerwork001@proton.me>
Date:   Tue May 21 09:53:35 2024 +0200

    MergeWithMain

### CODE REVIEW

**Comment by JustTheDude001** [comment]
Perdon con las diferencias.
Aqui queda más claro los cambios.
.env                                               |   11 +
 .env.docker                                        |    4 +
 .github/workflows/deploy.yml                       |   15 +
 .gitignore                                         |    1 -
 Makefile                                           |    2 +
 .../authAnnotations/AnnotationsAuth.php            |   84 +-
 app/DTO/UserDTO.php                                |   15 +
 app/Http/Controllers/api/AuthController.php        |   61 +
 app/Http/Controllers/api/LoginController.php       |    0
 app/Http/Requests/SigninRequest.php                |   38 +
 app/Service/User/UserService.php                   |  123 +
 composer.json                                      |    1 +
 composer.lock                                      |   63 +-
 config/app.php                                     |    1 +
 config/database.php                                |   22 +
 database/seeders/SigninTestSeeder.php              |   40 +
 docker-compose.yml                                 |   13 +
 docker/.gitignore                                  |    3 +-
 docker/redis/.conf/redis.conf                      | 1320 +++++++++
 routes/api/v1.php                                  |    2 +
 storage/api-docs/api-docs.json                     | 2886 --------------------
 tests/Feature/SigninMockeryTest.php                |  230 ++
 tests/Feature/SigninTest.php                       |  118 +
 tests/Unit/Service/UserServiceTest.php             |  291 ++
 24 files changed, 2454 insertions(+), 2890 deletions(-)

**Comment by jordimorillo** [comment]
Tengo varias observaciones:

- Si yo fuese un usuario que viene de fuera del contexto del proyecto y leyese lo que pone en esta PR, todavia no tendría claro qué va a pasar si ésta PR se mezcla con main. Es importante indicar ésto ya que no todos los que lean la PR se acordarán de dónde estamos ubicados.

- Hay conflictos. Cuando se presenta una rama para revisar tiene que estar libre de ellos. Para ello siempre es importante antes de hacer la PR, hacer un ultimo merge con main.

- Todos los checks tienen que pasar. Comprueba en local que todos los tests te pasan. Si fallan en producción pero en local no, levanta la mano.

File: .env.docker [hunk 1]
```diff
@@ -5,3 +5,7 @@ DB_PORT=3306
 DB_DATABASE=laravel
 DB_USERNAME=user
 DB_PASSWORD=password
+KEY_HASH_MAC=token_password
```
Comment (jordimorillo) [inline .env.docker:8]: look at docker-compose.yml because we are setting some environment variables there

**Comment by jordimorillo** [comment]
Merge main branch in yours because docker-compose is not updated.

**Pull request closed without merging**